src/pocketmine/utils/MainLogger.php

```diff
@@ -174,6 +174,19 @@ public function setLogDebug(bool $logDebug){
 	 * @param array|null $trace
 	 */
 	public function logException(\Throwable $e, $trace = null){
+		$generator = $this->logExceptionMessages($e, $trace, false);
+		$messages = iterator_to_array($generator, false);
```

Comment:
I'm not very clear why this is a generator to begin with, since it's only used here.

Reply:
You can use `array_push($target, ...$array2)`

Reply:
That would result in O(n^2) performance, which is at the same complexity class of array_merge(). See this benchmark (we are push-right here): https://gist.github.com/SOF3/988c408a92f79a644cc3fbc65727ca8d

Using array_unshift() is possible, but too complicated for such a simple feature where optimization isn't really critical. (You don't have an exception with 100 chained causes every day; if you do, you don't want to log it anyway)